python_modules/dagster/dagster/_core/definitions/composite_partitions.py

```diff
@@ -0,0 +1,39 @@
+import itertools
+from datetime import datetime
+from typing import Optional, Sequence
+
+from .partition import Partition, PartitionsDefinition
+
+
+class CompositePartitionsDefinition(PartitionsDefinition):
+    """The set of partitions is the cross product of partitions in the inner partitions
+    definitions"""
+
+    def __init__(self, partitions_defs: Sequence[PartitionsDefinition]):
+        self._partitions_defs = partitions_defs
+
+    @property
+    def partitions_defs(self):
+        return self._partitions_defs
+
+    def get_partitions(self, current_time: Optional[datetime] = None) -> Sequence[Partition]:
```

Comment:
I can't think of a reason why someone would want to do this, but do you think it's possible people might want to specify a different time for each partitions definition?